clang/test/CXX/special/class.temporary/p6.cpp

```diff
@@ -238,3 +239,236 @@ void init_capture_init_list() {
   // CHECK: call {{.*}}dtor
   // CHECK: }
 }
+
+namespace P2718R0 {
```

Comment:
Is it possible to also test the case where it does not clean up like in the proposal:

```cpp
for (auto e : f2(g())) {}  // undefined behavior
```

Reply:
agree, I have compiled this case, Currently, clang will not extending lifetime of temporaries in this case.